frontends/precinct-scanner/src/components/replace_ballot_bag_modal.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Text, Modal } from '@votingworks/ui';
+import { throwIllegalValue } from '@votingworks/utils';
+import { ScannedBallotCount } from './scanned_ballot_count';
+import { CenteredLargeProse } from './layout';
+import { BALLOT_BAG_CAPACITY } from '../config/globals';
+
+interface Props {
+  scannedBallotCount: number;
+  pollWorkerAuthenticated: boolean;
+  onComplete: VoidFunction;
+}
+
+type ModalState = 'no_auth' | 'confirming' | 'confirmed';
+
+export function ReplaceBallotBagModal({
+  scannedBallotCount,
+  pollWorkerAuthenticated,
+  onComplete,
+}: Props): JSX.Element {
+  const [modalState, setModalState] = useState<ModalState>('no_auth');
+
+  useEffect(() => {
+    if (modalState === 'no_auth' && pollWorkerAuthenticated) {
+      setModalState('confirming');
+    } else if (modalState === 'confirming' && !pollWorkerAuthenticated) {
+      setModalState('no_auth');
+    } else if (modalState === 'confirmed' && !pollWorkerAuthenticated) {
+      onComplete();
+    }
+  }, [modalState, onComplete, pollWorkerAuthenticated]);
+
+  const mainContent = (() => {
+    switch (modalState) {
+      case 'no_auth':
+        return (
+          <CenteredLargeProse>
+            <h1>The Ballot Bag is Full</h1>
```

Comment:
I think our other headings in VxScan don't use full sentences and articles (e.g.  "Ballot Not Counted" or "Too Many Votes"). To stick with that pattern, maybe just "Ballot Bag Full" here?

Reply:
Yeah, this would be more consistent. I had changed to this sentence to make it a bit more friendly/approachable in the case that a voter was the first to see it. Lets change it for consistency!